src/call.cpp

```diff
@@ -985,4 +985,46 @@ void DarlingServer::Call::S2CPerform::processCall() {
 	_sendReply(code);
 };
 
+void DarlingServer::Call::SetExecutablePath::processCall() {
+	int code = 0;
+	int fullLength = 0;
+
+	if (auto thread = _thread.lock()) {
+		if (auto process = thread->process()) {
+			std::string tmpstr;
+			tmpstr.resize(_body.buffer_size);
+			if (!process->readMemory((uintptr_t)_body.buffer, tmpstr.data(), _body.buffer_size, &code)) {
+				code = -code;
+			}
+			process->setExecutablePath(tmpstr.c_str());
+		} else {
+			code = -ESRCH;
+		}
+	} else {
+		code = -ESRCH;
+	}
+
+	_sendReply(code == 0 ? fullLength : code);
+}
+
+void DarlingServer::Call::GetExecutablePath::processCall() {
+	int code = 0;
+	uint64_t fullLength;
+
+	if (auto maybeTargetProcess = processRegistry().lookupEntryByNSID(_body.pid)) {
+		auto targetProcess = *maybeTargetProcess;
+		auto path = targetProcess->executablePath();
+		auto len = std::max(path.length() + 1, _body.buffer_size);
+		if (!targetProcess->writeMemory((uintptr_t)_body.buffer, path.data(), len, &code)) {
+			code = -code;
+		}
+		fullLength = path.length();
+		
+	} else {
+		code = -ESRCH;
```

Comment:
With darlingserver calls, negative error codes indicate internal errors, e.g. RPC communication failure or the current process or thread is missing (which should be impossible). However, it's perfectly reasonable for the caller to provide an invalid PID that doesn't exist, so this should be a positive error code.
```suggestion
		code = ESRCH;
```